### PR TITLE
feat: count method update for event analysis

### PR DESCRIPTION
**Pull request by llmin**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [x] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts [hunk 1]
```diff
@@ -116,19 +116,34 @@ export interface VisualRelatedDefProps {
...
+  readonly subTitle: string;
+  readonly tableTitle: string;
+}
+
+export interface DashboardDefProps {
+  def: DashboardVersionDefinition;
+  name?: string;
+}
+
 export type MustachePathAnalysisType = {
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts:130]: You should create a base type to share the common properties.

File: src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts [hunk 2]
```diff
@@ -818,4 +879,62 @@ export function getTempResourceName(resourceName: string, action: ExploreRequest
   }
 
   return resourceName;
-}
\ No newline at end of file
+}
+
+export function getDashboardTitleProps(analysisType: AnalysisType, query: any) : DashboardTitleProps {
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts:884]: let's use lib `i18next` to fetch the i18n string from external translation files.

File: src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts [hunk 3]
```diff
@@ -481,7 +489,7 @@ function _buildEventCondition(eventNames: string[], sqlParameters: SQLParameters
     ),
     `);
   }
-  return newSQL;
+  return [newSQL, computedMethodList];
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts:492]: return an object to improve the code readability 

```ts
return {
   sql: '',
  methodList: []
};
```

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts (hunk 2 above)
Comment (llmin) [reply src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts:897]: means we need initialize a i18n context in backend code ?

Comment (zxkane) [reply src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts:897]: see https://github.com/i18next/i18next-http-middleware

Comment (llmin) [reply src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts:897]: fixed

File: src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts (hunk 3 above)
Comment (llmin) [reply src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts:492]: fixed

File: src/control-plane/backend/lambda/api/test/api/reporting.test.ts [hunk 4]
```diff
@@ -582,11 +601,19 @@ describe('reporting test', () => {
...
+      Analysis: {
+        Name: 'test-analysis',
+      },
+    });
 
     const res = await request(app)
       .post('/api/reporting/path')
       .send({
         action: 'PUBLISH',
+        language: 'EN',
```
[the remark this answers is not in the document]
Comment (llmin) [reply src/control-plane/backend/lambda/api/test/api/reporting.test.ts:614]: fixed

**Review by zxkane — CHANGES_REQUESTED** [review]
- resolve conflicts locally

File: frontend/src/pages/analytics/analytics-utils.ts [hunk 5]
```diff
@@ -443,3 +444,8 @@ export const getWarmUpParameters = (
     },
   };
 };
+
+export const getLngFromLocalStorage = () => {
```
[the remark this answers is not in the document]
Comment (tyyzqmf) [reply frontend/src/pages/analytics/analytics-utils.ts:447]: return i18nextLng directly.

**Review by zxkane — APPROVED** [review]

**Pull request merged**